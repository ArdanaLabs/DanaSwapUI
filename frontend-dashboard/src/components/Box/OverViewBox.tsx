import React from "react"
import cx from "classnames"
import { Box, useMediaQuery, Typography } from "@material-ui/core"
import { makeStyles, useTheme } from "@material-ui/core/styles"

import * as Theme from "Data/User/Theme"

import { useUserTheme } from "state/user/hooks"

import CircleInfoCyanIcon from "assets/imgs/circle-info-cyan.svg"
import CircleInfoDarkBlueIcon from "assets/imgs/circle-info-darkblue.svg"

const useStyles = makeStyles(({ palette }) => ({
  root: {
    width: "100%",
    position: "relative",
    background: palette.background.paper,
    borderRadius: "10px",
    height: 100,
  },
  leftBorder: {
    position: "absolute",
    left: 0,
    top: 0,
    bottom: 0,
    height: "100%",
    background: `linear-gradient(180deg, ${palette.secondary.main} 0%, ${palette.secondary.dark} 100%)`,
    borderRadius: "10px",
  },
  info: {
    position: "absolute",
    top: 3,
    right: 10,
  },
  display: {
    padding: "25px 0px 10px 20px",

    [`& p:first-child`]: {
      color: palette.secondary.main,
      whiteSpace: "pre-line",
      textTransform: "uppercase",
      lineHeight: "100%",
      marginBottom: "10px",
    },
    [`& p:last-child`]: {
      color: palette.primary.main,
    },
  },
}))

export interface OverViewBoxProps {
  label: string
  content: string
}

const OverViewBox: React.FC<OverViewBoxProps> = ({ label, content }) => {
  const { breakpoints } = useTheme()
  const userTheme: Theme.Theme = useUserTheme()
  const mobile = useMediaQuery(breakpoints.down("xs"))
  const isDarkTheme: boolean = Theme.Eq.equals(userTheme, Theme.Theme.Dark)
  const classes = useStyles({ dark: isDarkTheme, mobile })

  return (
    <Box className={cx(classes.root)}>
      <Box className={cx(classes.leftBorder)}>&nbsp;&nbsp;</Box>
      <Box className={cx(classes.info)}>
        <img
          src={isDarkTheme ? CircleInfoCyanIcon : CircleInfoDarkBlueIcon}
          alt="info"
        />
      </Box>

      <Box className={cx(classes.display)}>
        <Typography variant="body2" component="p">
          {label}
        </Typography>
        <Typography variant="subtitle2" component="p">
          {content}
        </Typography>
      </Box>
    </Box>
  )
}

export default OverViewBox
