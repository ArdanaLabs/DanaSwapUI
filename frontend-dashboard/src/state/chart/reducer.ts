import {
  ActionReducerMapBuilder,
  PayloadAction,
  createReducer,
} from "@reduxjs/toolkit"

import * as E from "fp-ts/Either"
import { RemoteData, fromEither, pending } from "fp-ts-remote-data"

import {
  receivedAggVolume,
  receivedAggLiquidity,
  receivedPoolFees,
  receivedPoolVolume,
  receivedPoolLiquidity,
  receivedPoolTxCount,
  receivedPoolAPY,
  receivedPoolTransactions,
} from "./actions"

import {
  APYChart,
  FeeVolumeChart,
  LiquidityChart,
  TxCountChart,
  VolumeChart,
} from "Data/Chart"
import { FetchDecodeError, FetchDecodeResult } from "Data/FetchDecode"
import * as Transaction from "Data/Transaction"

export type State = {
  aggLiquidity: RemoteData<FetchDecodeError, LiquidityChart.Type>
  aggVolume: RemoteData<FetchDecodeError, VolumeChart.Type>
  poolFees: RemoteData<FetchDecodeError, FeeVolumeChart.Type>
  poolVolume: RemoteData<FetchDecodeError, VolumeChart.Type>
  poolLiquidity: RemoteData<FetchDecodeError, LiquidityChart.Type>
  poolTxCount: RemoteData<FetchDecodeError, TxCountChart.Type>
  poolAPY: RemoteData<FetchDecodeError, APYChart.Type>
  poolTransactions: RemoteData<FetchDecodeError, Transaction.WithTotalValue[]>
}

export const initialState: State = {
  aggLiquidity: pending,
  aggVolume: pending,
  poolFees: pending,
  poolVolume: pending,
  poolLiquidity: pending,
  poolTxCount: pending,
  poolAPY: pending,
  poolTransactions: pending,
}

// side-effectful: function that logs on Left, and returns the typical fromEither
// TODO: move this to chain with the decoder stuff
function fromEitherLogFail<A>({
  payload,
}: PayloadAction<FetchDecodeResult<A>>): RemoteData<FetchDecodeError, A> {
  if (E.isLeft(payload)) {
    console.error("Fetch error:", payload.left)
  }
  return fromEither(payload)
}

export default createReducer(
  initialState,
  (builder: ActionReducerMapBuilder<State>) =>
    builder
      .addCase(receivedAggLiquidity, (state, action) => ({
        ...state,
        aggLiquidity: fromEitherLogFail(action),
      }))
      .addCase(receivedAggVolume, (state, action) => ({
        ...state,
        aggVolume: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolFees, (state, action) => ({
        ...state,
        poolFees: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolVolume, (state, action) => ({
        ...state,
        poolVolume: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolLiquidity, (state, action) => ({
        ...state,
        poolLiquidity: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolTxCount, (state, action) => ({
        ...state,
        poolTxCount: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolAPY, (state, action) => ({
        ...state,
        poolAPY: fromEitherLogFail(action),
      }))
      .addCase(receivedPoolTransactions, (state, action) => ({
        ...state,
        poolTransactions: fromEitherLogFail(action),
      }))
)
