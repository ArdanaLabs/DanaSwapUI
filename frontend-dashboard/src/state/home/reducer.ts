import {
  ActionReducerMapBuilder,
  PayloadAction,
  createReducer,
} from "@reduxjs/toolkit"

import { RemoteData, fromEither, pending } from "fp-ts-remote-data"

import { CombinedStats } from "Data/Stats/CombinedStats"
import { FetchDecodeError, FetchDecodeResult } from "Data/FetchDecode"

import { receivedCombinedStats } from "./actions"

export type State = RemoteData<FetchDecodeError, CombinedStats>

export const initialState = pending

export default createReducer(
  initialState,
  (builder: ActionReducerMapBuilder<State>) =>
    builder.addCase(
      receivedCombinedStats,
      (
        _state: State,
        action: PayloadAction<FetchDecodeResult<CombinedStats>>
      ) => fromEither(action.payload)
    )
)
