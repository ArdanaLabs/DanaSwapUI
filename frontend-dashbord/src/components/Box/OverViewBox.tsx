import React from "react";
import cx from "classnames";
import { Box, useMediaQuery } from "@material-ui/core";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import { useIsDarkMode } from "state/user/hooks";

import ICO_Info_light from "assets/svg/info_light.svg";
import ICO_Info_dark from "assets/svg/info_dark.svg";

const useStyles = makeStyles(({ palette }) => ({
  self: {
    margin: "10px",
  },
  bg: {
    width: "100%",
    position: "relative",
    background: palette.background.paper,
    borderRadius: "10px",
    height: 100,
  },
  leftBorder: {
    position: "absolute",
    left: 0,
    top: 0,
    bottom: 0,
    height: "100%",
    background: palette.secondary.light,
    borderRadius: "10px",
  },
  info: {
    position: "absolute",
    top: 0,
    right: 10,
  },
  display: {
    padding: "10px 0px 10px 30px",

    "& p:first-child": {
      fontWeight: 300,
      fontStyle: "normal",
      fontSize: "13px",
      lineHeight: "13px",
      color: palette.text.hint,
    },
    "& p:last-child": {
      fontWeight: 700,
      fontStyle: "normal",
      fontSize: "18px",
      lineHeight: "18px",
      color: palette.text.secondary,
      fontFamily: "Museo Sans",
    },
  },
}));

export interface OverViewBoxProps {
  label: string;
  content: string;
  info?: string;
}

const OverViewBox: React.FC<OverViewBoxProps> = ({ label, content, info }) => {
  const { breakpoints } = useTheme();
  const dark = useIsDarkMode();
  const mobile = useMediaQuery(breakpoints.down("xs"));
  const classes = useStyles({ dark, mobile });

  return (
    <Box className={cx(classes.self)}>
      <Box
        className={cx(classes.bg)}
        style={
          !dark
            ? {
                boxShadow: "0px 4px 4px #E5E5E5",
              }
            : {
                boxShadow: "unset",
              }
        }
      >
        <Box className={cx(classes.leftBorder)}>&nbsp;&nbsp;</Box>
        <Box className={cx(classes.info)}>
          <img src={!dark ? ICO_Info_light : ICO_Info_dark} alt="info" />
        </Box>

        <Box className={cx(classes.display)}>
          <Box component="p">{label}</Box>
          <Box component="p">{content}</Box>
        </Box>
      </Box>
    </Box>
  );
};

export default OverViewBox;
