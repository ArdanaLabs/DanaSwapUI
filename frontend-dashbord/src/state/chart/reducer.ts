import { createReducer } from "@reduxjs/toolkit";

import { updateAggVol } from "./actions";

export const initialState: any = {
  aggVolsPerFiveMins: [],
};

export default createReducer(initialState, (builder) =>
  builder.addCase(updateAggVol, (state, action) => {
  })
);
