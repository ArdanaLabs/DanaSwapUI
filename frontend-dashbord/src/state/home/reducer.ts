import { createReducer } from "@reduxjs/toolkit";

import { TotalStat, updateTotalStats } from "./actions";

export const initialState: TotalStat = {
  totalDepositsAllPoolsUSD: 0,
  totalDailyVolumeUSD: null,
  totalDailyTxCount: 0,
  totalDailyFeeVolumeUSD: 0,
  totalLiquidityUtilization: 0,
  poolStats: null,
};

export default createReducer(initialState, (builder) =>
  builder.addCase(updateTotalStats, (state, action) => {
    const {
      totalDepositsAllPoolsUSD,
      totalDailyVolumeUSD,
      totalDailyTxCount,
      totalDailyFeeVolumeUSD,
      totalLiquidityUtilization,
      poolStats,
    } = action.payload;

    state.totalDepositsAllPoolsUSD = totalDepositsAllPoolsUSD;
    state.totalDailyVolumeUSD = totalDailyVolumeUSD;
    state.totalDailyTxCount = totalDailyTxCount;
    state.totalDailyFeeVolumeUSD = totalDailyFeeVolumeUSD;
    state.totalLiquidityUtilization = totalLiquidityUtilization;

    state.poolStats = poolStats;
  })
);
