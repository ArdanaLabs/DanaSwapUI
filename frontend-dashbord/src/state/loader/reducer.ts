import { createReducer } from "@reduxjs/toolkit";

import { setLoading } from "./actions";

export const initialState = {
  loading: false
};

export default createReducer(initialState, (builder) =>
  builder.addCase(setLoading, (state, action) => {
    state.loading = action.payload;
  })
);
