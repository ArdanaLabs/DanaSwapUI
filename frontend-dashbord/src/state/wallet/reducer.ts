import { createReducer } from "@reduxjs/toolkit";

import { getAccountKeys } from "./actions";

export interface WalletStateType {
  privateKey: string | null;
  publicKey: string | null;
  rewardAddress: string | null;
  accountId: string | null;
  accountIdFull: string | null;
}

export const initialState: WalletStateType = {
  privateKey: null,
  publicKey: null,
  rewardAddress: null,
  accountId: null,
  accountIdFull: null,
};

export default createReducer(initialState, (builder) =>
  builder.addCase(getAccountKeys, (state, action) => {
    state.privateKey = action.payload.privateKey;
    state.publicKey = action.payload.publicKey;
    state.rewardAddress = action.payload.rewardAddress;
    state.accountId = action.payload.accountId;
    state.accountIdFull = action.payload.accountIdFull;
  })
);
