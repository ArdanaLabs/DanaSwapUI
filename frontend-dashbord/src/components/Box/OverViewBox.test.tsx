import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import * as Enzyme from 'enzyme'
import configureMockStore, { MockStoreEnhanced } from 'redux-mock-store'
import { ReactWrapper, shallow, mount } from "enzyme";
import { Provider, useSelector } from 'react-redux';
import { initialState } from 'state/user/reducer';
import { OverViewBox } from ".";
import { OverViewBoxProps } from "./OverViewBox";
import { Box } from '@material-ui/core';

Enzyme.configure({ adapter: new Adapter() })

const mockDispatch = jest.fn()
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn(f => f())
}))
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch
}))
const mockStore = configureMockStore([])

let store: MockStoreEnhanced<unknown>;

describe("Components / Box / OverViewBox", () => {
  beforeEach(() => {
    store = mockStore(initialState)
    ;(useSelector as jest.Mock).mockImplementation(callback => {
      return callback(initialState)
    })
  })

  afterEach(() => {
    ;(useSelector as jest.Mock).mockClear()
  })

	it('renders', () => {
    const wrapper = shallow(<Provider store={store}><OverViewBox label={"TVL\n\n"} content={"$220.21 M"} /></Provider>)
    expect(wrapper).toMatchSnapshot();
	})
});
