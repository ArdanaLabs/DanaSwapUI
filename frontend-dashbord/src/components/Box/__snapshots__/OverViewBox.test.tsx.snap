// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Box / OverViewBox renders 1`] = `ShallowWrapper {}`;
